src/core.jl

```diff
@@ -90,7 +90,11 @@ function Base.run(x::JSONRPCEndpoint)
 
     x.write_task = @async try
         for msg in x.out_msg_queue
-            write_transport_layer(x.pipe_out, msg)
+            if isopen(x.pipe_out)
+                write_transport_layer(x.pipe_out, msg)
+            else
+                break
```

Comment:
```suggestion
                # TODO Reconsider at some point whether this should be treated as an error.
                break
```

I'm not sure, at the moment, and I think we should merge this because it might solve a problem. At the same time, I'm not sure whether we still have something in place then that would report an error condition in some way.